File: ios/Classes/TaskContext.m
//
// Created by Razvan Lung(long1eu) on 2019-02-20.
// Copyright (c) 2019 The Chromium Authors. All rights reserved.
//

#import "TaskContext.h"


@implementation TaskContext
- (instancetype)initWithArguments:(id)arguments resultHandler:(ResultHandler)resultHandler {
    self = [super init];
    if (self) {
        self.arguments = arguments;
        self.resultHandler = resultHandler;
    }

    return self;
}

@end

File: ios/Classes/TaskContext.swift
//
//  TaskContext.swift
//  geolocator
//
//  Created by Maurits van Beusekom on 03/07/2018.
//

import Foundation

typealias ResultHandler = (_ result: Any?) -> ()

class TaskContext : NSObject {
    
    init(resultHandler: @escaping ResultHandler, arguments: Any?) {
        self.resultHandler = resultHandler
        self.arguments = arguments
    }
    
    let arguments: Any?
    let resultHandler: ResultHandler
}
